### DISCUSSION TITLE
Why is my GitHub Actions workflow stuck in the “queued” state?

### DISCUSSION THREAD

**Discussion by anotheradamibnu** [community/community#147604] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi everyone,

I’m trying to run a GitHub Actions workflow on my repository, but it remains in the “queued” state and doesn’t proceed to execution. I’ve already checked the following:
	1.	There are no pending jobs ahead of it in the queue.
	2.	I haven’t exceeded the usage limits for my account or repository.
	3.	The runner for the job is correctly specified and should be available.

Does anyone know what might be causing this issue or how I can debug it further?

Thanks in advance!

**Comment by Adamibnualfatahh** [comment, marked as answer]
This issue can happen for several reasons. Here are a few things you can check to fix the “queued” state problem:
	1.	Check Runner Availability
Ensure that the runner you specified in your workflow file is available and not occupied by other jobs. You can check the status of your self-hosted runners in the repository settings under Actions > Runners. If you’re using GitHub-hosted runners, they might be busy with other jobs, especially during high demand periods.
	2.	Job Dependencies
If your workflow includes jobs that depend on others, make sure those jobs have completed successfully before the queued job starts. Sometimes, if a job dependency fails or hasn’t started, it can cause subsequent jobs to stay in the “queued” state.
	3.	Check Usage Limits
If you’re on a free plan or have hit usage limits, this can cause jobs to be queued until capacity is available. You can review the usage limits for GitHub Actions in your account settings to make sure you’re within the limits.
	4.	Cancel Stuck Jobs
Sometimes a stuck job can cause other jobs to get queued indefinitely. If this is the case, try canceling the queued job and re-running it. This can sometimes resolve the issue.
	5.	GitHub Status
Lastly, check [GitHub’s status page](https://www.githubstatus.com/) to see if there are any ongoing issues with GitHub Actions or infrastructure that might be causing delays.

Let me know if any of these steps help!

**Reply by investincommerce** [reply]
Thanks — I'd already worked through all five before writing in, and none apply here:
• No self-hosted runners (0 configured); runs-on: ubuntu-latest, and other runs in this same repository pick up hosted runners within seconds — a build succeeded today at 13:38 UTC.
• The workflow has a single job with no needs: dependencies.
• Not a usage limit — other runs in this repo and org complete normally.
• Cancelling is the thing that fails: both cancel and force-cancel return HTTP 500 "Failed to cancel workflow run". DELETE returns 403, and rerun returns 403 "This workflow is already running".
• githubstatus.com shows Actions operational, and these have been stuck for over four days.

The key detail: GET /actions/runs/<id>/jobs returns {"total_count": 0} for all five. No job was ever created, so there is nothing queued for a runner — these are orphaned run records, created while Actions was disabled at the repo level (since re-enabled). They cannot be reached by any API call.

**Comment by IndianPhotoAcademy** [comment]
When a GitHub Actions workflow is stuck in the "queued" state, common causes include a lack of available runners, reaching repository limits, or running too many workflows at once. Check the Actions settings, make sure the runners are configured, and double-check your plan's limits. Cancel unnecessary workflows to make room for pending ones.

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @anotheradamibnu!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Actions category. I've gone ahead and moved it for you. Good luck!

**Comment by LiamSwayne** [comment]
I am having a similar issue but all my actions are frozen as queued across all repos. Any reason this may be happening? It seems to have happened overnight and is not the result of any code changes.

**Comment by user-Roma** [comment]
@LiamSwayne , having the same issue today evening, thats why im here.
Looks like smthng on github side

**Reply by LiamSwayne** [reply]
Thanks! Thought I was tripping when I woke up and every CI/CD pipeline I ever created was frozen.

**Reply by user-Roma** [reply]
I tried to rerun pipelines for working in the past deployments and they also having status Queued.
I thought that maybe i could reach some limits but it was first run during last 3 days..

**Reply by petedanielsmith** [reply]
mine was queued for an hour, i quit it thinking it was my fault and repushed and its still an hour and a half queued now. def an issue their end, although status page still shows no issues anywhere

**Reply by LiamSwayne** [reply]
mine started working 25 seconds ago

**Reply by user-Roma** [reply]
for me still not working

**Reply by stampd-dev** [reply]
not working again for me today

**Comment by LiamSwayne** [comment]
@GitHub please look into this it has crashed production for me and probably a lot of other people that haven't noticed yet

**Comment by NikitaVelixJob** [comment]
bump, same issue Queued

**Comment by iampraveen08** [comment]
The issue of a GitHub Actions workflow getting stuck in the “queued” state can be caused by several factors. Here are some steps you can take to troubleshoot and resolve the problem:

Verify Runner Availability
Ensure that the runner specified in your workflow is available.

For self-hosted runners, go to your repository settings → Actions > Runners and check their status.

For GitHub-hosted runners, delays can happen during periods of high demand. Try re-running the job after some time.

Check Job Dependencies
If your workflow contains multiple jobs with dependencies (defined using needs:), confirm that the prerequisite jobs have completed successfully. If a dependent job hasn't started or has failed, the next job may remain queued.

Review Usage Limits
Ensure that you haven't hit GitHub Actions usage limits. This includes limits on minutes, storage, or concurrent jobs—especially if you're using a free or lower-tier plan. You can monitor usage in your account or organization settings.

Cancel and Retry
Sometimes a job can become stuck in the queue indefinitely. In such cases, cancel the job and trigger it again. This often helps clear the queue state.

Check GitHub Status
Visit the [GitHub Status Page](https://www.githubstatus.com/) to see if there are any incidents or degraded performance affecting GitHub Actions. Platform-wide issues can cause unexpected delays.

**Reply by petedanielsmith** [reply]
I'm well below usage limits, I tried restarting, it's clearly an issue as it's happening to multiple people. The status page is no help as just says everything is fine. Now been over 12 hours stuck in queue and GitHub aren't even reporting an issue

**Reply by NikitaVelixJob** [reply]
i restarted my job and now its worked

**Reply by petedanielsmith** [reply]
I did that when it first got stuck for an hour. Just reluctant to try again if it means I get stuck in another 12 hour queue at the back again 😀 maybe I'll give it a go

**Reply by petedanielsmith** [reply]
Finally it's working again!

**Reply by user-Roma** [reply]
@iampraveen08 , my pipelines also started working working by itself few minutes ago, after another page refresh

**Comment by Prath-0506** [comment]
1.	Check GitHub Status:Visit [status.github.com](https://www.githubstatus.com/) to see if GitHub Actions runners are experiencing outages or high load.
2.	Are you using GitHub-hosted runners (like ubuntu-latest)?
 If yes — they should auto-scale.
 Are you using a self-hosted runner?
 Make sure it’s online and shows as Idle in your repo’s Settings → Actions → Runners.
3.	Double-check your runs-on in the workflow YAML. It must match the runner’s labels exactly (for example: runs-on: [self-hosted, linux]).
4.	If it’s a PR from a fork, do you see an “Approve and run” button in the Actions tab?.Someone with write access must approve it.
5.	Are there other jobs ahead in the queue?. Sometimes concurrency or limited runners can create a hidden backlog.
6.	Visit your repo/org’s Actions Usage settings to see if you hit free minutes or storage limits.
7.	Do you use concurrency in your YAML?.If so, a previous run in the same group might be blocking new runs.
8.	Is your runs-on or matrix syntax valid?. Even small typos can prevent a job from matching any runner.
9.	If it’s an org repo, see if there are policies restricting runner groups. Example: your repo may not have permission to use certain runners.
10.	Click the workflow run → click the queued job → see if you get hints like “Waiting for runner to come online.”
11.	Sometimes the runner process has crashed or disconnected.Stop & restart it, or re-register if needed.
12.	Create a new workflow that just echoes Hello World. This helps rule out YAML issues.
13.	Temporarily change runs-on to a different OS/version. Example: ubuntu-latest instead of ubuntu-22.04.
14.	For private repos, confirm you have Actions minutes left.Free accounts have limited free minutes.
15.	If you’ve tried everything and it stays queued for hours, [open a support ticket](https://support.github.com/).
16.	Most commonly, it’s either:
•	The runner is offline/unavailable,
•	The labels don’t match,
•	Or you need to approve a workflow from a fork.

**Comment by ghost** [comment]
Similar issue. pages-build-deployment is stuck for more than 6 hours
https://github.com/lingonas/lingonas.github.io/actions/runs/16104288239

Tried "Cancel run", but it returns "Failed to cancel workflow."

**Reply by F3rrY** [reply]
Same here, right now

**Comment by abrehamgezahegn** [comment]
Experiencing the same issue here.

**Comment by w4ss1mIBM** [comment]
Experiencing the same issue here.

**Comment by nghhoese** [comment]
same issue here suddenly some workflows wont run anymore and stay queud for forever canceling and restarting dont seem to work. 
Thought i was going crazy before finding this thread.

**Comment by abinasharma001** [comment]
### 💬 **Answer:**

Hi! You're doing the right checks already — thanks for sharing the context. If your GitHub Actions workflow is stuck in the **“queued”** state, but nothing obvious is blocking it, here are some additional things to verify:

---

### ✅ **Troubleshooting Checklist**

#### 1. 🧠 **Check for Runner Availability**

Even if you’ve specified a runner, make sure that:

* The **runner is online and idle** (for self-hosted).
* You’re targeting the correct **label** (`ubuntu-latest`, `self-hosted`, etc.).
* For **enterprise or org-hosted runners**, check if the runner group has **permission** for the repo.

> **Tip:** Go to **Repository Settings → Actions → Runners** to check runner status.

---

#### 2. 📦 **Workflow Concurrency Limits**

Check for `concurrency` settings in your workflow YAML. If you’ve configured it like:

```yaml
concurrency:
  group: my-job
  cancel-in-progress: true
```

...then another run might be **blocking** it without being obvious.

---

#### 3. 🔒 **Repository Permissions or Secrets**

If the workflow uses **secrets** or **environments with protection rules**, GitHub might delay or pause the job until approval or access is granted.

---

#### 4. ⚙️ **GitHub Actions System Issues**

Check:

* [[GitHub Status](https://www.githubstatus.com/)](https://www.githubstatus.com/)
* [[Actions incident history](https://www.githubstatus.com/history)](https://www.githubstatus.com/history)
  GitHub sometimes has runner-related incidents that can delay job starts.

---

#### 5. 🪵 **Enable Debug Logging**

Add the following to your workflow or repo secrets for more logging:

```env
ACTIONS_RUNNER_DEBUG: true
ACTIONS_STEP_DEBUG: true
```

---

### 🛠️ Still Not Running?

If it stays queued for >5–10 minutes with no logs or movement:

* Cancel and re-run the job.
* Try triggering it from another branch.
* Post a screenshot or YAML snippet here for more targeted help.

---

Hope this helps you get unstuck! Let me know what your `runs-on` and `concurrency` values look like if you're still blocked.

**Reply by LiamSwayne** [reply]
Bro this is AI-generated lmao pls stop. For people wondering why this is happening it's probably actions being over capacity. Even on the pro plan I was not able to get anything to run.

**Reply by petedanielsmith** [reply]
Think it was definitely a bug. I found if I replied on a push triggering it that it went to queue and stayed there, sometimes for like 12 hours, but if I cancelled that job and reran failed jobs it worked but a fresh push would go back in the stuck in queue status. Rerunning just 1 failed job seemed to also get it to pick up other jobs I had stuck in queue too. Not tried again today though

**Reply by LiamSwayne** [reply]
Yea that sounds like a bug

**Comment by grant-zukowski-xumo** [comment]
For me I was gone for awhile and I had my runner version os to ~ubuntu 20.  There was no error but it hung on queue.  I bumped the runner version to 22.04 and that got it unstuck.

I hope this helps someone.

**Reply by GajaNen** [reply]
Yep, I was in the same situation and changing to 22.04 worked., thanks!

**Reply by akochepasov** [reply]
This is the solution. They stopped supporting old OS runners

**Reply by ajimix** [reply]
This also worked for me 🙏

**Comment by pipedexplumbing** [comment]
Same issue.

**Comment by james-harlyy** [comment]
Happening for me, starting today. Not a usage issue, we've used less for 5% of our Enterprise plan usage. All the workflows are stuck in "queued" (for about an hour now)

**Comment by s-titov** [comment]
Same here, can't deploy to prod, just infinite Queued. I've tried re-run but it didn't help.

**Comment by Sewer56** [comment]
I ran into this problem recently.
When I contacted GitHub Support, the person at the other end was helpful;
and they also let me know of a certain API after helping me out:

- [Force cancel a workflow run](https://docs.github.com/en/enterprise-cloud@latest/rest/actions/workflow-runs?apiVersion=2022-11-28#force-cancel-a-workflow-run) (See the docs for more details)

There's a chance this might help you folks.

**Comment by sramekmichal** [comment]
If you get something like:
```
{
  "message": "Cannot cancel a workflow re-run that has not yet queued.",
  "documentation_url": "https://docs.github.com/rest/actions/workflow-runs#force-cancel-a-workflow-run",
  "status": "409"
}
```

Just run in your repo:
```
git commit --allow-empty -m "trigger re-run"
git push
```

It is gonna skip the previous jobs in stuck state and run fresh job.

**Reply by L0g0ff** [reply]
This works! Thanks!

**Reply by jalbarranelectronic** [reply]
Some years later, this tip saved my life. Thanks!

**Reply by tolzy88** [reply]
I owe you a beer!

**Comment by Vishnupriya-Anupindi** [comment]
I am having the same problem now, don't know what to do. I tried cancelling the workflow and to Re-run all jobs. I added the following to the yml file and changed the node version from 16.x to 20.x but even that job is Queued. 
concurrency:
  group: build
  cancel-in-progress: true

**Comment by jellytha** [comment]
[]()

**Comment by jellytha** [comment]
[]()

**Comment by Bluesboy** [comment]
Same issue here, can't even cancel the workflow, UI gives says "Failed to cancel workflow.", CLI says "Cannot cancel a workflow run that is completed". None of methods in the accepted answer work.

**Comment by Gutza** [comment]
I am having the same problem. I have four instances of the same action on a GitHub pages repo, all stuck in _Queued_ state, with `Failed to cancel workflow.` if I try to cancel them.

For my use case, it's obvious this is a bug: this is a simple Jekyll build-and-deploy action with three tasks:
1. Build the Jekyll website
2. Report the build status
3. Deploy the website

The dependencies are (1) -> (2) and (1) -> (3) – notably, **not** (1) -> (2) -> (3).

All four instances have tasks (1) and (2) completed successfully, and task (3) completed, timed out, or canceled – and yet **the actions are all Queued!**

So this is the perfect Catch-22:
- Attempting to cancel the run doesn't make sense, since no task is running (nor is it really queued to run, since all actions are in final states) – so the error "Failed to cancel workflow." is legitimate (there's nothing to cancel);
- Starting a new action fails, because there are "queued" actions which also attempt to deploy to the same website;
- However, because of the bug, timed out deployment tasks don't error the action instance, they leave it queued indefinitely.

**Comment by Gutza** [comment]
This fixed it for me: https://github.com/orgs/community/discussions/49074#discussioncomment-15489079

**Comment by leoswing** [comment]
Stuck in the queued state for now, the jobs just queued and could not run any more .. May I know any tips about that?

**Reply by ShelleeNikole** [reply]
There is an active GitHub actions outage

https://www.githubstatus.com
